tracing-journald/src/memfd.rs

```diff
@@ -8,7 +8,18 @@ use std::os::raw::c_uint;
 use std::os::unix::prelude::{FromRawFd, RawFd};
 
 fn create(flags: c_uint) -> Result<File> {
-    let fd = unsafe { memfd_create("tracing-journald\0".as_ptr() as *const c_char, flags) };
+    // Make the `memfd_create` syscall ourself instead of going through `libc`;
+    // `memfd_create` isn't supported on `glibc<2.27` so this allows us to
+    // support old-but-still-used distros like Ubuntu Xenial, Debian Stretch,
+    // RHEL 7, etc.
+    // See: https://github.com/tokio-rs/tracing/issues/1879
+    let fd = unsafe {
+        syscall(
```

Comment:
Let's wrap the syscall in a dedicated safe function perhaps?  That'd make this a bit easier to read 🙂

Reply:
Done